polaris/hub/client.py

```diff
@@ -617,8 +617,8 @@ def upload_dataset(
                 dest.store[".zmetadata"] = zmetadata_content
 
                 logger.info("Copying Zarr archive to the Hub. This may take a while.")
-                zarr.copy_all(source=dataset.zarr_root, dest=dest, log=logger.info)
-
+                zarr.copy_store(source=dataset.zarr_root.store.store, dest=dest.store, log=logger.info, if_exists='replace')
```

Comment:
I'm not sure about setting the `if_exists` parameter here. I think all values (replace, skip, raise) can be useful. For example, if you accidentally stop your download, you'll likely want to use "skip" to continue where you left off. 

Maybe add it as a parameter to the `upload_dataset method` and keep it as "raise" by default?

Reply:
@cwognum Yesterday I noticed that during upload, I was adding (i.e., replacing in this case) .zgroup to the same path twice. Just raising this because I would then assume that most, if not all, archives on initial upload might hit this error.

Reply:
Hm, good catch! Maybe that it's because when we open the Zarr archive, it already creates some files?

Maybe we shouldn't open the archive, but just create the respective store object. Could you try replacing [this line of code](https://github.com/polaris-hub/polaris/blob/main/polaris/dataset/_dataset.py#L351):
```py
dest = zarr.open(zarr_archive, "w")
```

by something like: 
```py
dest = zarr.DirectoryStore(zarr_archive)
```

(similar for the upload process)